core/Test/Tasty/Run.hs

```diff
@@ -118,10 +119,16 @@ executeTest action statusVar timeoutOpt inits fins = mask $ \restore -> do
     -- anyway.
     initResources
 
+    let
+      cursorMischiefManaged = do
+        -- Could this be better managed? I feel like it could ... 
```

Comment:
Any conclusion about this TODO?

Reply:
That's a question to @mankyKitty.

Reply:
I must confess all memory of this change has left my mind. To hazard a guess I think I was unsure that using a TVar was  the best choice. If wires became crossed or this wasn't managed correctly, tests could start blocking one another waiting to update a `statusVar` when they should have exited/completed.

I don't have a concrete answer for you though, so you may disregard/remove at your discretion.